xpath.js

```diff
@@ -280,7 +280,8 @@ var xpath = (typeof exports === 'undefined') ? {} : exports;
         };
         this.reduceActions[28] = function (rhs) {
             rhs[0].locationPath = rhs[2];
-            rhs[0].locationPath.steps.unshift(new Step(Step.DESCENDANTORSELF, NodeTest.nodeTest, []));
+            rhs[0].locationPath.steps.push(new Step(Step.DESCENDANTORSELF, NodeTest.nodeTest, []));

```

Comment:
Seems like a change in behavior. Need to figure out what's going on here.

Reply:
This would result in a change in functionality because you don't control the incoming array. For example, if the `steps` array is [1, 2, 3, 4] and you `unshift(5)` you get `[5, 1, 2, 3, 4]`. If you `push(5)` and then `reverse` it you get `[5, 4, 3, 2, 1]`. It is only appropriate to `push` and `reverse` when you control the start and end of the array's creation, or you can `reverse` it before you start processing. It doesn't appear that either of those apply in this case.

Reply:
Looking at the other cases in this PR, you should be good. Just in this case the simple course would be to revert this change due to not controlling the start and end of array creation.